### ISSUE TITLE
Nano: inference pipeline metric evaluation sometimes may bring some wrong answers

### ISSUE THREAD

**Issue by TheaperDeng** [intel-analytics/BigDL#5589]
As said in this TODO, this implementation is only valid for dataloader with all batchsizes are the same

https://github.com/intel-analytics/BigDL/blob/1b45bf03830cd5c47edaf23bd78705edd5143313/python/nano/src/bigdl/nano/pytorch/inference/optimizer.py#L436-L444

**Comment by jason-dai** [comment]
Maybe we should check and report error (instead of failing silently)

**Comment by TheaperDeng** [comment]
> Maybe we should check and report error (instead of failing silently)

I think we need to fix it directly since 1. this bug will only affect minorly to the final metric value 2. it happens very frequently since sample_num/batch_size is not often to be an integer.

Some possible implementation can be:
```python
def _accuracy_calculate_helper(model, metric, data): 
     sample_num_count = 0
     # ...
     for i, (data_input, target) in enumerate(data): 
        # ...
        sample_num_count += data_input.shape[0]
    return np.sum(metric_list) / sample_num_count
```
